src/main/java/tools/jackson/databind/MapperFeature.java

```diff
@@ -443,7 +443,18 @@ public enum MapperFeature
      *
      * @since 2.13
      */
-    APPLY_DEFAULT_VALUES(true)
+    APPLY_DEFAULT_VALUES(true),
+
+    /**
+     * Feature that determines whether {@link ObjectWriter} and {@link ObjectWriter} unwraps
+     * the root cause of exception before wrapping with Jackson exceptions.
+     *<p>
+     * This is useful to keep the stack trace of the original exception, when customized
+     * exceptions are thrown in extension points such as setter, constructor, creator, etc.
+     *<p>
+     * Feature is enabled by default.
+     */
+    UNWRAP_ROOT_CAUSE(true)
```

Comment:
Since this is for 3.0, I think we can default it to `false`.

Reply:
I personally don't see the value of having this configurable in the first place - for me, I would always be OK with nested exceptions where Jackson adds helpful descriptions like `JacksonMappingException: couldn't write field 'foo' from type 'bar'` but always preserves the cause `caused by: IllegalAccessError: your module doesn't read the other module` or whatever it is.

I would always keep the ultimate cause myself, even if it is not "friendly". But I can understand how other people might have different expectations, like Spring Boot will try to hide the failure details from you behind some friendly message - but I hate it ;)

Reply:
Thanks @stevenschlansker. I'll need to think about this a bit... I am not married to the unwrapping feature even tho I  originally added it in the first place.